connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -903,69 +903,73 @@ export async function microsoftGarbageCollectionActivity({
         n.nodeType === "file"
     );
 
-  // Explicitly not using concurrency here to avoid too many calls to the api
-  // This is a long running workflow that does not require high throughput
-  for (const node of nodesToCheck) {
-    const { itemAPIPath } = typeAndPathFromInternalId(node.internalId);
-
-    let driveOrItem: DriveItem | undefined | Drive = undefined;
-    try {
-      driveOrItem = await getItem(client, itemAPIPath);
-    } catch (e) {
-      // if the entity is not found, we can go on and delete
-      if (!(e instanceof GraphError && e.statusCode === 404)) {
-        throw e;
-      }
-    }
+  const requests = nodesToCheck.map((n, i) => ({
+    id: `${i}`,
+    url: typeAndPathFromInternalId(n.internalId).itemAPIPath,
+    method: "GET",
+  }));
 
-    switch (node.nodeType) {
-      case "drive":
-        if (!driveOrItem || !rootNodeIds.includes(node.internalId)) {
-          await deleteFolder({ connectorId, internalId: node.internalId });
-        }
-        break;
-      case "folder": {
-        const folder = driveOrItem as DriveItem;
-        if (
-          !folder ||
-          folder.deleted ||
-          // isOutsideRootNodes
-          (await isOutsideRootNodes({
-            client,
-            driveItem: folder,
-            rootNodeIds,
-            startGarbageCollectionTs,
-          }))
-        ) {
-          await deleteFolder({ connectorId, internalId: node.internalId });
-        }
-        break;
-      }
-      case "file": {
-        const file = driveOrItem as DriveItem;
-        if (
-          !file ||
-          file.deleted ||
-          // isOutsideRootNodes
-          (await isOutsideRootNodes({
-            client,
-            driveItem: file,
-            rootNodeIds,
-            startGarbageCollectionTs,
-          }))
-        ) {
-          await deleteFile({
-            connectorId,
-            internalId: node.internalId,
-            dataSourceConfig,
-          });
+  const chunkedRequests = [];
+  while (requests.length) {
+    chunkedRequests.push(requests.splice(0, 20));
+  }
+
+  for (const chunk of chunkedRequests) {
+    const batchRes = await client.api("/$batch").post({ requests: chunk });
```

Comment:
just checking that when you tested in local the replies were correct

Reply:
yes i tried with existing files, and removed files (not sure if I went into 404 or deleted flag, but the file was garbage collected)